src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/QuestionManager.java

```diff
@@ -465,14 +452,28 @@ private void shuffleChoiceQuestionsChoices(final String seed, final List<Questio
         // shuffle all choices based on the seed provided, augmented by individual question ID.
         for (QuestionDTO question : questions) {
             if (question instanceof ChoiceQuestionDTO) {
+                ChoiceQuestionDTO choiceQuestion = (ChoiceQuestionDTO) question;
+                String qSeed = seed + choiceQuestion.getId();
+
                 Boolean randomiseChoices = ((ChoiceQuestionDTO) question).getRandomiseChoices();
                 if (randomiseChoices == null || randomiseChoices) {  // Default to randomised if not set.
-                    ChoiceQuestionDTO choiceQuestion = (ChoiceQuestionDTO) question;
-                    String qSeed = seed + choiceQuestion.getId();
                     if (choiceQuestion.getChoices() != null) {
                         Collections.shuffle(choiceQuestion.getChoices(), new Random(qSeed.hashCode()));
                     }
                 }
+
+                // FIXME: this is an Isaac specific thing in a segue class!
+                // Perhaps ItemQuestions could live in Segue, but then what relation should they have to
+                // the IsaacQuestionBase class?
```

Comment:
However item questions do seem like something which would be useful for any other application that would be built on top of Segue. IsaacQuestionBaseDTO doesn't currently seem to be used anywhere and I don't really see a reason where it would be useful to have it so maybe the answer is to remove it.